processing/src/main/java/org/apache/druid/math/expr/vector/CastToStringVectorProcessor.java

```diff
@@ -25,18 +25,20 @@
 
 public final class CastToStringVectorProcessor extends CastToTypeVectorProcessor<Object[]>
 {
-  public CastToStringVectorProcessor(ExprVectorProcessor<?> delegate)
+  private final Object[] output;
+
+  public CastToStringVectorProcessor(ExprVectorProcessor<?> delegate, int maxVectorSize)
   {
     super(delegate);
+    this.output = new Object[maxVectorSize];
   }
 
   @Override
   public ExprEvalVector<Object[]> evalVector(Expr.VectorInputBinding bindings)
   {
     ExprEvalVector<?> result = delegate.evalVector(bindings);
     final Object[] objects = result.getObjectVector();
-    final Object[] output = new String[objects.length];
-    for (int i = 0; i < objects.length; i++) {
+    for (int i = 0; i < bindings.getCurrentVectorSize(); i++) {
```

Comment:
was this a bug?

Reply:
I don't think so, previously it would just cast the whole vector size instead of only what was needed